File: UI/warehouse-management/src/app/components/warehouses/warehouse.ts
import { Product } from '../all-products/product';

export interface NewWarehouse {
  name: string;
  maxCapacity: number;
  currentCapacity: number; // Zero if empty, equal to maxCapacity if full
  fullness?: number;
  country: string;
  zipcode?: string;
  city?: string;
  address?: string;
}

export interface Warehouse extends NewWarehouse {
  warehouseId: string;
}

export interface WarehouseProduct {
  warehouse: Warehouse;
  amount: number;
}

File: backend/WarehouseManagement.API/Models/Domain/Warehouse.cs
namespace WarehouseManagement.API.Models.Domain
{
    public class Warehouse
    {
        public Guid WarehouseId { get; set; }
        public required string Name { get; set; }
        public required int MaxCapacity { get; set; }
        // CurrentCapacity is 0 when empty, equal to MaxCapacity if WH is full.
        public int CurrentCapacity { get; set; }

        // Location attributes
        public required string Country { get; set; }
        public string? Zipcode { get; set; }
        public string? City { get; set; }
        public string? Address { get; set; }
    }
}
